internal/states/statemgr/helper.go

```diff
@@ -45,7 +45,7 @@ func RefreshAndRead(mgr Storage) (*states.State, error) {
 // is required, call WriteState and PersistState on the state manager directly
 // and handle their errors.
 func WriteAndPersist(mgr Storage, state *states.State) error {
```

Comment:
@uturunku1 I wonder if we should also accept schema here and pass through to WriteState. That would feel more consistent with the interface as well as make a couple of caller migrations unnecessary.